### PR TITLE
[WIP]Refactor vit2

### PR DESCRIPTION
**Pull request by HIT-cwh**
Hi! This pr refactors vision transformer with FFN and MSA in transformer.py, mmcv. As the refactor_tramsformer_base branch created by shilong isn't merged now, all the unitests failed.
Pls kindly take a look.

### CODE REVIEW

File: mmcls/models/backbones/vision_transformer.py [hunk 1]
```diff
@@ -1,217 +1,79 @@
...
+        drop_path_rate (float): stochastic depth rate. Default 0.0
+        num_fcs (int): The number of fully-connected layers for FFNs. Default 2
+        qkv_bias (bool): enable bias for qkv if True. Default True
+        act_cfg (dict): The activation config for FFNs. Defalut GELU
         norm_cfg (dict): Config dict for normalization layer. Default
-            layer normalization.
-        num_fcs (int): The number of fully-connected layers for FFNs.
-            Default 2.
+            layer normalization
+        batch_first (bool): Key, Query and Value are shape of
```
Comment (Junjun2016) [inline mmcls/models/backbones/vision_transformer.py:29]: Make clearer description about `batch_first`.
It is hard to follow, maybe add why and common cases in CV would be better.

File: mmcls/models/backbones/vision_transformer.py [hunk 2]
```diff
@@ -1,217 +1,79 @@
...
                  embed_dims,
                  num_heads,
                  feedforward_channels,
-                 attn_drop=0.,
-                 proj_drop=0.,
+                 drop_rate=0.,
+                 attn_drop_rate=0.,
+                 drop_path_rate=0.,
+                 num_fcs=2,
+                 qkv_bias=True,
```
Comment (Junjun2016) [inline mmcls/models/backbones/vision_transformer.py:43]: It seems that the default `qkv_bias` is `False` (`qkv_bias=False`) in most repos (e.g. [TIMM](https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/vision_transformer.py#L136)), so maybe we can follow the common practices.

File: mmcls/models/backbones/vision_transformer.py (hunk 1 above)
Comment (HIT-cwh) [reply mmcls/models/backbones/vision_transformer.py:29]: Good point. Fixed.

File: mmcls/models/backbones/vision_transformer.py (hunk 2 above)
Comment (sennnnn) [reply mmcls/models/backbones/vision_transformer.py:43]: In fact, the default value of qkv_bias is True. (e.g. [TIMM](https://github.com/rwightman/pytorch-image-models/blob/07fb05cc3d5c6f99d5b74a88e9a14db42e8e2e4f/timm/models/vision_transformer.py#L193))

File: mmcls/models/backbones/vision_transformer.py [hunk 3]
```diff
@@ -297,10 +165,15 @@ def forward(self, x):
...
+
         return x
 
 
-class HybridEmbed(nn.Module):
+# Modified from pytorch-image-models
+class HybridEmbed(BaseModule):
     """CNN Feature Map Embedding.
 
     Extract feature map from CNN, flatten, project to embedding dim.
```
Comment (Junjun2016) [inline mmcls/models/backbones/vision_transformer.py:179]: Docstring of `Args`

File: mmcls/models/backbones/vision_transformer.py [hunk 4]
```diff
@@ -254,8 +117,10 @@ def __init__(self,
                  patch_size=16,
                  in_channels=3,
                  embed_dim=768,
-                 conv_cfg=None):
-        super(PatchEmbed, self).__init__()
+                 norm_cfg=None,
+                 conv_cfg=None,
+                 init_cfg=None):
```
Comment (Junjun2016) [inline mmcls/models/backbones/vision_transformer.py:122]: Add `kernel_size`,  `stride` and `padding`  args to suppport overlap embedding.

**Comment by mzr1996** [comment]
Closed, continue with #395
